diff --git a/packages/flutter/lib/src/widgets/selectable_region.dart b/packages/flutter/lib/src/widgets/selectable_region.dart
index 59de8bae20b..07ac6981c78 100644
--- a/packages/flutter/lib/src/widgets/selectable_region.dart
+++ b/packages/flutter/lib/src/widgets/selectable_region.dart
@@ -2822,7 +2822,10 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
       return null;
     }
     final buffer = StringBuffer();
-    for (final selection in selections) {
+    for (final (i, selection) in selections.indexed) {
+      if (i != 0) {
+        buffer.write('\n');
+      }
       buffer.write(selection.plainText);
     }
     return SelectedContent(plainText: buffer.toString());
